torch/distributed/nn/api/remote_module.py

```diff
@@ -143,6 +151,19 @@ def __init__(
             method = torch.jit.export(method)
             setattr(self, method_name, types.MethodType(method, self))
 
+    def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef[Parameter]]:
+        r"""Returns a list of RRefs of remote module parameters.
+        This is typically passed to a distributed optimizer.
+        Args:
+            recurse (bool): if True, then returns parameters of the remote module
+                and all submodules of the remote module.
+                Otherwise, returns only parameters that are direct members of the remote module.
+
+        Returns:
+            A list of RRefs to remote module parameters.
+        """
+        return rpc.rpc_sync(self.on, _param_rrefs, args=(self.module_rref, recurse))
```

Comment:
Just curious, would it be good to support an async API for this as well?  It could be useful if users want to retrieve the parameters for several remote modules and an async API would allow us to do this concurrently.